runelite-client/src/main/java/net/runelite/client/plugins/poh/PohPlugin.java

```diff
@@ -0,0 +1,122 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.poh;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.api.GameObject;
+import net.runelite.api.GameState;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameObjectDespawned;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.Overlay;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+@PluginDescriptor(
+	name = "Poh plugin"
+)
+public class PohPlugin extends Plugin
+{
+	@Getter(AccessLevel.PACKAGE)
+	private final Set<GameObject> pohObjects = new HashSet<>();
+
+	@Inject
+	PohOverlay overlay;
+
+	@Inject
+	BurnerOverlay burnerOverlay;
+
+	@Inject
+	PohConfig config;
+
+	@Override
+	public void configure(Binder binder)
+	{
+		binder.bind(PohOverlay.class);
+	}
+
+	@Provides
+	PohConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(PohConfig.class);
+	}
+
+	@Override
+	public Collection<Overlay> getOverlays()
+	{
+		return Arrays.asList(overlay, burnerOverlay);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlay.updateConfig();
+	}
+
+	@Subscribe
+	public void updateConfig(ConfigChanged event)
+	{
+		overlay.updateConfig();
+	}
+
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		GameObject gameObject = event.getGameObject();
+		if (Stream.of(burnerOverlay.getBurnerLit(), burnerOverlay.getBurnerUnlit()).anyMatch(x -> x.contains(gameObject.getId())) || PohIcons.getIcon(gameObject.getId()) != null)
```

Comment:
just check for lit and unlit burners respectively, no need to create streams on the fly. Also, the line is way too long (I think, GitHub is not showing how long